frame/session/src/lib.rs

```diff
@@ -647,7 +653,9 @@ impl<T: Config> Pallet<T> {
 		let session_keys = <QueuedKeys<T>>::get();
 		let validators =
 			session_keys.iter().map(|(validator, _)| validator.clone()).collect::<Vec<_>>();
-		Validators::<T>::put(&validators);
+
+		let validator2: BoundedVec<T::ValidatorId, T::ValidatorSet> = validators.clone().try_into().expect("Too many current validators");
```

Comment:
You certainly cannot and should not panic here -- need to handle it defensively, either use `WeakBoundedVec`, or truncate to fit (better).

Reply:
Resolved?

Reply:
not yet -- you cannot panic in any runtime code path such as on_initialize or a transaction.

Reply:
Mapping to a custom error would be ideal?